src/samplers/tcp/stat.rs

```diff
@@ -0,0 +1,141 @@
+    #[strum(serialize = "tcp/connect/latency")]
+    ConnectLatency,
+    #[strum(serialize = "tcp/receive/segment")]
+    ReceiveSegments,
+    #[strum(serialize = "tcp/transmit/segment")]
+    TransmitSegments,
+    #[strum(serialize = "tcp/receive/prune_called")]
+    ReceivePruneCalled,
+    #[strum(serialize = "tcp/receive/collapsed")]
+    ReceiveCollapsed,
+    #[strum(serialize = "tcp/transmit/retransmit")]
+    Retransmits,
+    #[strum(serialize = "tcp/receive/checksum_error")]
+    ReceiveChecksumErrors,
+    #[strum(serialize = "tcp/transmit/reset")]
+    TransmitResets,
+    #[strum(serialize = "tcp/receive/error")]
+    ReceiveErrors,
+    #[strum(serialize = "tcp/syncookies/sent")]
+    SyncookiesSent,
+    #[strum(serialize = "tcp/syncookies/received")]
+    SyncookiesRecieved,
+    #[strum(serialize = "tcp/syncookies/failed")]
+    SyncookiesFailed,
+    #[strum(serialize = "tcp/receive/pruned")]
+    ReceivePruned,
+    #[strum(serialize = "tcp/receive/ofo_pruned")]
+    ReceiveOfoPruned,
+    #[strum(serialize = "tcp/transmit/delayed_ack")]
+    TransmitDelayedAcks,
+    #[strum(serialize = "tcp/receive/listen_overflows")]
+    ReceiveListenOverflows,
+    #[strum(serialize = "tcp/receive/listen_drops")]
+    ReceiveListenDrops,
+}
+
+impl TcpStatistic {
+    pub fn keys(self) -> Option<(&'static str, &'static str)> {
+        match self {
+            Self::ReceiveSegments => Some(("Tcp:", "InSegs")),
+            Self::TransmitSegments => Some(("Tcp:", "OutSegs")),
+            Self::ReceivePruneCalled => Some(("TcpExt:", "PruneCalled")),
+            Self::ReceiveCollapsed => Some(("TcpExt:", "TCPRcvCollapsed")),
+            Self::Retransmits => Some(("Tcp:", "RetransSegs")),
+            Self::ReceiveChecksumErrors => Some(("Tcp:", "InCsumErrors")),
+            Self::TransmitResets => Some(("Tcp:", "OutRsts")),
+            Self::ReceiveErrors => Some(("Tcp:", "InErrs")),
+            Self::SyncookiesSent => Some(("TcpExt:", "SyncookiesSent")),
+            Self::SyncookiesRecieved => Some(("TcpExt:", "SyncookiesRecv")),
+            Self::SyncookiesFailed => Some(("TcpExt:", "SyncookiesFailed")),
+            Self::ReceivePruned => Some(("TcpExt:", "RcvPruned")),
+            Self::ReceiveOfoPruned => Some(("TcpExt:", "OfoPruned")),
+            Self::TransmitDelayedAcks => Some(("TcpExt:", "DelayedACKs")),
+            Self::ReceiveListenOverflows => Some(("TcpExt:", "ListenOverflows")),
+            Self::ReceiveListenDrops => Some(("TcpExt:", "ListenDrops")),
+            _ => None,
+        }
+    }
+
+    pub fn ebpf_table(self) -> Option<&'static str> {
+        match self {
+            Self::ConnectLatency => Some("connlat"),
+            _ => None,
+        }
+    }
+}
+
+impl Statistic for TcpStatistic {
+    fn name(&self) -> &str {
+        (*self).into()
+    }
+
+    fn description(&self) -> Option<&str> {
+        Some(match self {
+            Self::ConnectLatency => "latency of active tcp connect",
```

Comment:
Something that's not for this PR, but would be neat - generating the docs from these strings. Maybe have this as part of the --help output (or behind a different flag?) and generate from that.

Reply:
Definitely not for this PR, but I was trying to figure if the build.rs could generate the METRICS.md. at the very least, some help text and http exposition are possible. Open to suggestions on strategy here. I would rather maintain in one location, but a build.rs strategy wasn't obvious when I looked into it.